apps/web/src/services/useHttp.ts

```diff
@@ -0,0 +1,79 @@
+import { Dispatch, SetStateAction, useCallback, useEffect, useState } from 'react';
+import { AxiosRequestConfig } from 'axios';
+import { toast } from 'react-toastify';
+import { AxiosPublic } from '../utils';
+import { REQUEST_METHOD } from '../common';
+
+type HttpReturn = {
+  sendApiRequest: any;
+  setIsLoading: Dispatch<SetStateAction<boolean>>;
+  isLoading: boolean;
+  fetchData: any;
+};
+
+export const useHttp = (): HttpReturn => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const controller = new AbortController();
+
+  const errorHandler = (err: any) => {
+    if (err?.response?.status === 400) {
+      toast.error({
+        status: 'error',
+        message: 'Kindly verify the input',
+      });
+    } else {
+      toast.error({
+        status: 'error',
+        message: err?.response?.data?.message ?? 'Error fetching data',
+      });
+    }
+  };
+
+  const fetchData = useCallback(async (requestConfig, handleData) => {
```

Comment:
@leon-fw Can you use this hook to make get api calls

Reply:
Sure thing, looks great!